src/useEventListener/useEventListener.spec.ts

```diff
@@ -0,0 +1,158 @@
+import { fireEvent, render } from '@testing-library/vue'
+import { vi, describe, it, test, expect, afterEach } from 'vitest'
+import { ref } from 'vue'
+import { useEventListener } from '@/useEventListener/useEventListener'
+import { timeout } from '@/utilities/tests'
+import * as utils from '@/utilities/vue'
+
+describe('useEventListener', () => {
+
+  afterEach(() => {
+    vi.restoreAllMocks()
+  })
+
+  test.each<null | undefined>([
+    undefined,
+    null,
+  ])('given falsy target never adds event listener', (initialValue) => {
+    const callback = vi.fn()
+    const target = ref<HTMLElement | undefined | null>(initialValue)
+    vi.spyOn(utils, 'toValue').mockReturnValue(initialValue)
+
+    useEventListener(target, 'click', callback)
+
+    if (target.value) {
+      const addEventListenerMock = vi.spyOn(target.value, 'addEventListener')
+      expect(addEventListenerMock).not.toBeCalled()
+    }
```

Comment:
yeah, this was a bit awkward because the test is currently setup in a way where this would never be executed but it felt like a way of explaining functionality. I don't think it provides much value, I will remove